raft-tech/TANF-app
# feat(1387): As a System Admin, I should not be able to set conflicting STT and region attributes for a user

## Summary of Changes
_Provide a brief summary of changes_
Pull request closes #1387 
_Acceptance criteria as stated in the issue_

## How to Test
_List the steps to test the PR_
These steps are generic, please adjust as necessary.
```
cd tdrs-frontend && docker-compose -f docker-compose.yml -f docker-compose.local.yml up -d
cd tdrs-backend && docker-compose -f docker-compose.yml -f docker-compose.local.yml up -d 
```

1. Open http://localhost:3000/ and sign in.
2. Open django admin panel at http://locahost:8080/admin/
![image](https://user-images.githubusercontent.com/10676925/151394815-5a2d4fa1-3b5a-4dbe-8db5-5491c2b5ec8f.png)

3. navigate to users and select the use you want to change
![image](https://user-images.githubusercontent.com/10676925/151394904-3200c9d7-0616-4f9f-8130-4b979a65359c.png)

4. Assign the user to either the data analyst or regional staff group, and try to give the user an in appropriate location type

5. Attempt to save, you should see an error message
﻿![image](https://user-images.githubusercontent.com/10676925/151391555-c42a3ecb-4d00-44cb-b9fb-839496509b19.png)
## Deliverables
_More details on how deliverables herein are assessed included [here](https://github.com/raft-tech/TANF-app/blob/raft-tdp-main/docs/How-We-Work/our-priorities-values-expectations.md#Deliverables)._

### [Deliverable 1: Accepted Features](https://github.com/raft-tech/TANF-app/blob/raft-tdp-main/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-1-Accepted-Features)

Checklist of ACs:

- [x] Users cannot be created through the admin panel with both an STT and a region assigned by System Admin
- [x] STT Data Analysts will have permissions based on their STT location
- [x] OFA Regional Staff will have permissions based on their region location
- [ ] The admin panel does not have options for system administrator to try to create a user
_doesnt look like this was addressed based on screenshot for test # 3 above. also below_:
<img width="1072" alt="1387" src="https://user-images.githubusercontent.com/63075587/154534822-8eabcf3a-6b98-4cd1-af4f-ecf6436d21bf.png">

- [x] All other roles should have a null location

### [Deliverable 2: Tested Code](https://github.com/raft-tech/TANF-app/blob/raft-tdp-main/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-2-Tested-Code)

+ Are all areas of code introduced in this PR meaningfully tested?
+ Are code coverage minimums met?
  + [x] Frontend coverage: [_93.92%_] (see `CodeCov Report` comment in PR)
  + [x] Backend coverage: [_97.14%_] (see `CodeCov Report` comment in PR)

### [Deliverable 3: Properly Styled Code](https://github.com/raft-tech/TANF-app/blob/raft-tdp-main/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-3-Properly-Styled-Code)


### [Deliverable 4: Accessible](https://github.com/raft-tech/TANF-app/blob/raft-tdp-main/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-4-Accessibility)

_n/a_

### [Deliverable 5: Deployed](https://github.com/raft-tech/TANF-app/blob/raft-tdp-main/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-5-Deployed)

_`adpennington` tested this locally given migration hiccups_

### [Deliverable 6: Documented](https://github.com/raft-tech/TANF-app/blob/raft-tdp-main/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-6-Code-documentation)


### [Deliverable 7: Secure](https://github.com/raft-tech/TANF-app/blob/raft-tdp-main/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-7-Secure)

_no new issues detected_
_[backend report](https://16525-281707402-gh.circle-artifacts.com/0/tdrs-backend/reports/owasp_report.html) shows non-TDP URLs are being scanned, but this is being addressed as part of #1534_

### [Deliverable 8: User Research](https://github.com/raft-tech/TANF-app/blob/raft-tdp-main/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-8-User-Research)

_n/a_
